### DISCUSSION TITLE
Regression (?) for fetching release notes resulting in "No github.com token has been configured."

### DISCUSSION THREAD

**Discussion by malmor** [renovatebot/renovate#29220] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, using the docker image renovate/renovate:37.374.3

### Please tell us more about your question or problem

Hey everyone,
we started seeing some warnings during release note fetching - complaining about a missing github.com token. I believe this is a regression introduced by a recent code refactoring. This issue probably only occurs in some specific "edge" cases, see below.

### Current (faulty?) behaviour
Renovate throws warnings complaining about missing github.com tokens for fetching release notes:
```
WARN: No github.com token has been configured. Skipping release notes retrieval (repository=my-repo/git, branch=renovate/git-update)
       "manager": "regex",
       "packageName": "git",
       "sourceUrl": "https://git-for-windows.github.io/"
```
These errors are also displayed in the merge request.

### Expected (old) behaviour
The `sourceUrl` does not point to github.com but github.io instead - so I would expect renovate to just ignore this host when fetching release notes. In the past all merge requests updating this specific dependency just had no release notes (which was okay) - but now they contain warnings.

### Our setup
We are running the `renovate/renovate` docker image. During startup we fetch a github access token and provide it as `GITHUB_COM_TOKEN` (in addition to other secrets for internal registries).

There are no custom host rules or configurations regarding github or release notes fetching.

### Root cause / Regression
I believe https://github.com/renovatebot/renovate/pull/28945 introduced this change/regression.

Before the refactoring release notes were only fetched for github.com hosts:
https://github.com/renovatebot/renovate/blob/5c0628bf3b3b09ec91e68d555c51fc71559a2edd/lib/workers/repository/update/pr/changelog/github/source.ts#L59-L61

The refactoring changed that check to use `detectPlatform(sourceUrl)` instead:
https://github.com/renovatebot/renovate/blob/4a0ec6c92fc217f60cbce0458c6a8931945d48e1/lib/workers/repository/update/pr/changelog/github/source.ts#L45

https://github.com/renovatebot/renovate/blob/4a0ec6c92fc217f60cbce0458c6a8931945d48e1/lib/workers/repository/update/pr/changelog/github/source.ts#L58-L60

Looking at the implementation of `detectPlatform` for github not only github.com but any hostname that contains github is being matched / detected:
https://github.com/renovatebot/renovate/blob/4a0ec6c92fc217f60cbce0458c6a8931945d48e1/lib/util/common.ts#L36-L37

This way a `sourceUrl` like `https://git-for-windows.github.io/` is being recognized as `github` and release notes are being fetched.

This results in a follow-up issue:

Because we (and probably nobody else) do not provide an access token for github.io (just using the default `GITHUB_COM_TOKEN` that results in a host rule for github.com) fetching the release notes failes because the token is missing.

As a "quick fix" we just created a new host rule for github.io that uses the same access token:
```
hostRules: [
    {
      hostType: "github",
      matchHost: "github.io",
      token: process.env.GITHUB_COM_TOKEN,
      timeout: 30000
    }
]
```

This way the `No github.com token has been configured.` warning disappears but instead Renovate throws errors because github.io is actually not a valid github url:

```
Fetching changelog: https://git-for-windows.github.io/ (2.44.0 -> 2.45.1)
Invalid github URL found: https://git-for-windows.github.io/
```

### Reproduction

At the moment the affected repository is in our internal GitLab instance. But if it helps I could create a public reproduction in GitHub.

### Logs (if relevant)

<details><summary>Logs (without a host rule for github.io)</summary>

```
{"name":"renovate","level":20,"config":{"hostRules":[{"hostType":"github","matchHost":"github.com","token":"***********"}],"msg":"Env config"}
...
{"name":"renovate","level":20,"msg":"Adding token authentication for github.com (hostType=github) to hostRules"}
...
{"name":"renovate","level":20,"branch":"renovate/git-update","msg":"PR rebase requested, so skipping cache check"}
{"name":"renovate","level":20,"branch":"renovate/git-update","msg":"Fetching changelog: https://git-for-windows.github.io/ (2.44.0 -> 2.45.1)"}
{"name":"renovate","level":40,"branch":"renovate/git-update","manager":"regex","packageName":"git","sourceUrl":"https://git-for-windows.github.io/","msg":"No github.com token has been configured. Skipping release notes retrieval"}
...
```

</details>

<details><summary>Logs (with a custom host rule for github.io)</summary>

```
{"name":"renovate","level":20,"config":{"hostRules":[{"hostType":"github","matchHost":"github.com","token":"***********"}],"msg":"Env config"}
...
{"name":"renovate","level":20,"msg":"Adding token authentication for github.com (hostType=github) to hostRules"}
{"name":"renovate","level":20,"msg":"Adding token authentication for github.io (hostType=github) to hostRules"}
...
{"name":"renovate","level":20,"branch":"renovate/git-update","msg":"PR rebase requested, so skipping cache check"}
{"name":"renovate","level":20,"branch":"renovate/git-update","msg":"Fetching changelog: https://git-for-windows.github.io/ (2.44.0 -> 2.45.1)"}
{"name":"renovate","level":40,"branch":"renovate/git-update","msg":"Invalid github URL found: https://git-for-windows.github.io/"}
...
```

</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
Thanks for the detailed explanation. This change will be reverted https://github.com/renovatebot/renovate/pull/29255

**Reply by malmor** [reply]
Thank you @RahulGautamSingh - I can confirm that this has been fixed by the revert! 👍 

Closing this issue.
